spec/unit/resource/windows_firewall_rule_spec.rb

```diff
@@ -457,13 +491,92 @@
         resource.remote_port("8081")
         resource.direction(:outbound)
         resource.protocol("UDP")
+        resource.icmp_type("Any")
         resource.firewall_action(:notconfigured)
         resource.profile(:domain)
         resource.program('%WINDIR%\System32\lsass.exe')
         resource.service("SomeService")
         resource.interface_type(:remoteaccess)
         resource.enabled(false)
-        expect(provider.firewall_command("Set")).to eql("Set-NetFirewallRule -Name 'test_rule_the_second' -NewDisplayName 'some cool display name' -Description 'some other rule' -LocalAddress '192.168.40.40' -LocalPort '80' -RemoteAddress '8.8.4.4' -RemotePort '8081' -Direction 'outbound' -Protocol 'UDP' -Action 'notconfigured' -Profile 'domain' -Program '%WINDIR%\\System32\\lsass.exe' -Service 'SomeService' -InterfaceType 'remoteaccess' -Enabled 'false'")
+        expect(provider.firewall_command("Set")).to eql("Set-NetFirewallRule -Name 'test_rule_the_second' -NewDisplayName 'some cool display name' -Description 'some other rule' -LocalAddress '192.168.40.40' -LocalPort '80' -RemoteAddress '8.8.4.4' -RemotePort '8081' -Direction 'outbound' -Protocol 'UDP' -IcmpType 'Any' -Action 'notconfigured' -Profile 'domain' -Program '%WINDIR%\\System32\\lsass.exe' -Service 'SomeService' -InterfaceType 'remoteaccess' -Enabled 'false'")
+      end
+    end
+  end
+
+  describe "#is_set_properly?" do
+    context "#TCP" do
+      it "icmp_type is nil" do
+        expect(provider.is_set_properly?(nil, "TCP")).to eql(false)
+      end
+
+      it "icmp_type is empty" do
+        expect(provider.is_set_properly?("", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 'Any'" do
+        expect(provider.is_set_properly?("Any", "TCP")).to eql(true)
+      end
+
+      it "icmp_type is 123 as String" do
+        expect(provider.is_set_properly?("123", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 123 as Integer" do
+        expect(provider.is_set_properly?(123, "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 345 as String" do
+        expect(provider.is_set_properly?("345", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 345 as Integer" do
+        expect(provider.is_set_properly?(345, "TCP")).to eql(false)
+      end
+
+      it "icmp_type is '1:3' as code pair" do
+        expect(provider.is_set_properly?("1:3", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is '123:456' as code pair out of range" do
+        expect(provider.is_set_properly?("123:456", "TCP")).to eql(false)
+      end
```

Comment:
This probably needs a test for the opposite case, where the first number in the pair is out of range (ie, `456:123`).

Reply:
Right now, the check of the definition simply splits those pairs into single numbers and checks them, if any of these are not in a range. But if you want to have this test in here, I'll add it.